### ISSUE TITLE
Can't get past scan

### ISSUE THREAD

**Issue by steveschow** [apohl79/audiogridder#16]
Server won't get past scanning plugins.  It crashes after scanning for a long time and every time I restart it it tries to rescan the same long list of plugins again, crashing again, etc.. can't get past that phase.  The product should keep track of what has successfully scanned and proactively determine if something causes it to crash, skip scanning it (and all the others that already scanned) upon relaunch.

**Comment by apohl79** [comment]
It does, but something must go wrong. Can you provide the server log? It is located under ~/Library/Logs/AudioGridderServer/Main_*.log.

If you figure out what plugin(s) are causing crashes, your can also manually blacklist them. Open ~/.audiogridderserver or create the file:

```
{
    "AU": true,
    "BlacklistedPlugins": [
        "AudioUnit:Effects/aufx,AEqP,Eios",
        "/Library/Audio/Plug-Ins/VST3/Steinberg/Retrologue.vst3"
    ],
    "ExcludePlugins": [],
    "ID": 0,
    "VST": true
}
```
Put the plugins to be blacklisted into the "BlacklistedPlugins" list, and AG will skip them.

**Comment by steveschow** [comment]
Yes I can send you the log, it doesn't show much.  The last plugin scanned according to the log is **IK Multimedia's Lurssen Mastering Console**...but I reckon that was a successful scan and the _next_ one to be scanned is the one that is crashing...but I have no idea which one that is.

**Comment by steveschow** [comment]
And actually looking through the 15 or so log files...from 15 failed attempts to scan...its never the same plugin that is the last one on the list....  So honestly I have no idea how to identify which plugin is causing a problem

**Comment by apohl79** [comment]
Can you provide a stack trace of a server crash?

**Comment by steveschow** [comment]
please tell me where to find it.

**Comment by apohl79** [comment]
When it crashes, OSX shows a crash dialog. Please copy all info from that dialog into a text file and attach it.

**Comment by steveschow** [comment]
Its dying without putting up the crash dialog.

**Comment by apohl79** [comment]
I'm not sure if you familiar with lldb, but thats the point where you would run it in a debugger. Not much I can do from here. I'm sorry.

**Comment by steveschow** [comment]
I recommend you log something BEFORE each plugin is scanned so that we can easily see what was the last plugin that was scanning when it crashed.  And even better, upon restart, look for that log and blacklist that plugin automatically.  This is how every other DAW handles it.

**Comment by apohl79** [comment]
Yep, thats exactly what happens. Each plugin that will be scanned, will be logged to the logfile before and also be added to a special file. If the server crashes, it will blacklist that failing plugin. Thats why I have no idea whats going wrong on your machine.

**Comment by apohl79** [comment]
Closing this. If you have more info, it can be reopened.

**Comment by steveschow** [comment]
I don't have more info.  I gave up trying to use Audiogridder frankly.  The scan won't get through and no way to figure out why.  If and when it works better in the future I will try again.
